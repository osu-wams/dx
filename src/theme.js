export const colors = {
  // OSU Brand Colours
  orange: '#d73f09',
  black: '#000000',
  white: '#ffffff',
  pineStand: '#4a773c',
  highTide: '#00859b',
  luminance: '#ffb500',
  stratosphere: '#006a8e',
  reindeerMoss: '#c4d6a4',
  seafoam: '#b8dde1',
  candela: '#fdd26e',
  moondust: '#c6dae7',
  hopBine: '#aa9d2e',
  rogueWave: '#0d5257',
  solarFlare: '#d3832b',
  starCanvas: '003b5c',
  till: '#b7a99a',
  coastline: '#a7aca2',
  highDesert: '#7a6855',
  crater: '#8e9089',
  // Web errors, gray scales
  lava: '#af292e',
  vulcan: '#e69198',
  basalt: '#5e131a',
  mist: '#f0f0f0',
  fog: '#e0e0e0',
  overcast: '#999',
  nimbus: '#757575',
  dusk: '#444',
  charcoal: '#252525'
};

export const shadows = {
  1: '0px 1px 3px 0px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 2px 1px -1px rgba(0, 0, 0, 0.12)',
  2: '0px 1px 5px 0px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 3px 1px -2px rgba(0, 0, 0, 0.12)',
  3: '0px 1px 8px 0px rgba(0, 0, 0, 0.2), 0px 3px 4px 0px rgba(0, 0, 0, 0.14), 0px 3px 3px -2px rgba(0, 0, 0, 0.12)',
  4: '0px 2px 4px -1px rgba(0, 0, 0, 0.2), 0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12)',
  5: '0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 5px 8px 0px rgba(0, 0, 0, 0.14), 0px 1px 14px 0px rgba(0, 0, 0, 0.12)',
  6: '0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12)',
  7: '0px 4px 5px -2px rgba(0, 0, 0, 0.2), 0px 7px 10px 1px rgba(0, 0, 0, 0.14), 0px 2px 16px 1px rgba(0, 0, 0, 0.12)',
  8: '0px 5px 5px -3px rgba(0, 0, 0, 0.2), 0px 8px 10px 1px rgba(0, 0, 0, 0.14), 0px 3px 14px 2px rgba(0, 0, 0, 0.12)'
};

const theme = {
  default: {
    fg: colors.black,
    bg: colors.white
  },
  mist: {
    fg: colors.charcoal,
    bg: colors.mist
  },
  primary: {
    fg: colors.white,
    bg: colors.orange
  },
  secondary: {
    fg: colors.white,
    bg: colors.black
  },
  academic: {
    fg: colors.white,
    bg: colors.stratosphere
  },
  finance: {
    fg: colors.white,
    bg: colors.pineStand
  },
  spacing: {
    unit: 8
  },
  rounded: {
    normal: '0.8rem',
    subtle: '0.3rem'
  },
  fontSize: {
    small: '1.4rem',
    normal: '1.6rem',
    large: '1.8rem'
  },
  colors
};

export default theme;
