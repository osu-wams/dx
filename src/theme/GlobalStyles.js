import { createGlobalStyle } from 'styled-components';
import { themeSettings } from './';

const GlobalStyle = createGlobalStyle`
  html {
    box-sizing: border-box;
    font-size: 62.5%;
    font-family: 'Open Sans', sans-serif;
    *, *:before, *:after {
      box-sizing: inherit;
    }
  }
  body {
    padding: 0;
    margin: 0;
    font-size: ${themeSettings.fontSize[16]};
    line-height: 1.5;
    color: ${({ theme }) => theme.body.color};
  }
  body:not(.user-is-tabbing) button:focus,
  body:not(.user-is-tabbing) input:focus,
  body:not(.user-is-tabbing) select:focus,
  body:not(.user-is-tabbing) textarea:focus {
    outline: none;
  }
  #root {
    background-color: ${({ theme }) => theme.body.background};
    min-height: 100vh;
    display: flex;
    flex-direction: column;
  }
  input, select, textarea, button {
    font-size: inherit;
  }
  strong {
    font-weight: 600;
  }
  /* Router helper */
  .router-styles {
    display: flex;
    height: 100%;
    flex-direction: column;
  }
  @media screen and (max-width: 767px) {
    [data-reach-dialog-overlay] {
      background: ${({ theme }) => theme.ui.myDialog.background} !important;
      z-index: 11;
    }
  }
`;

export default GlobalStyle;
