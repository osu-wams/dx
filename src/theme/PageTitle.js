import styled from 'styled-components';

const PageTitle = styled.h1`
  font-family: Stratum2Web, sans-serif;
  font-size: 2rem;
  padding: 0 1.4rem;
  color: #555;
`;

export default PageTitle;
