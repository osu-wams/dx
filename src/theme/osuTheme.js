import { createMuiTheme } from '@material-ui/core/styles';

const osuTheme = createMuiTheme({
  palette: {
    primary: {
      // light: will be calculated from palette.primary.main,
      main: '#d73f09',
      strat: '#006A8E',
      white: '#fff'
      // dark: will be calculated from palette.primary.main,
      // contrastText: will be calculated to contrast with palette.primary.main
    },
    secondary: {
      main: '#555',
      light: '#f2f2f2',
      // dark: will be calculated from palette.secondary.main,
      contrastText: '#ffffff'
    },
    error: {
      main: '#000'
    }
    // error: will use the default color
  },
  typography: {
    fontFamily: '"Open Sans", "sans-serif"'
  },
  overrides: {
    // Permanently override card styles
    MuiCardHeader: {
      root: {
        backgroundColor: '#f2f2f2',
        paddingTop: '8px',
        paddingBottom: '8px'
      },
      title: {
        color: '#252525',
        fontSize: '1.4rem'
      },
      action: {
        color: '#d73f09',
        marginTop: '0',
        alignSelf: 'auto'
      }
    }
  }
});

export default osuTheme;
