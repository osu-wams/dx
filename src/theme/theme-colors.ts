export enum Color {
  'transparent' = 'transparent',
  'white' = '#ffffff',
  'black' = '#000000',
  // Neutral Palette
  'neutral-100' = '#F7F5F5', // only use as an off white background
  'neutral-200' = '#E9E5E4', // use as a light gray background with dark text
  'neutral-300' = '#D4CFCD', // use as a gray background with very dark text
  'neutral-400' = '#B7B1AF', // use carefully
  'neutral-500' = '#8F8582', // minimum contrast for dark ui (non-text) elements on white - WCAG 2.1
  'neutral-550' = '#7B746F', // minimum contrast for dark text with white background
  'neutral-600' = '#696361', // minimum for dark text with background up to neutral-200
  'neutral-700' = '#423E3C', // use as a text color for backgrounds as dark as neutral-400
  'neutral-800' = '#2E2B2A',
  'neutral-900' = '#1F1D1C',
  // Primary
  'orange-100' = '#FFE7DE',
  'orange-200' = '#F79572',
  'orange-300' = '#E85F2E',
  'orange-400' = '#D73F09', // Base shade
  'orange-500' = '#B22F00',
  'orange-600' = '#7D2100',
  'orange-700' = '#611A00',
  // Accents
  'stratosphere-100' = '#ECF1F4',
  'stratosphere-200' = '#C6DAE7',
  'stratosphere-300' = '#2490B5',
  'stratosphere-400' = '#006A8E', // Base shade
  'stratosphere-500' = '#005875',
  'stratosphere-600' = '#003B5C',
  'stratosphere-700' = '#002235',
  'pine-100' = '#EDF3E3',
  'pine-200' = '#C4D6A4',
  'pine-300' = '#73A663',
  'pine-400' = '#4A773C', // Base shade
  'pine-500' = '#3C692E',
  'pine-600' = '#254A1A',
  'pine-700' = '#142E0C',
  'lava-400' = '#C72127',
  'luminance-300' = '#FFCE57',
  'luminance-400' = '#FFB500',
  'roguewave-400' = '#00859B'
}
