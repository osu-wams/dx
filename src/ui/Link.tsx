import React, { useContext } from 'react';
import { Link } from '@reach/router';
import { faLongArrowRight, faExternalLink } from '@fortawesome/pro-light-svg-icons';
import Icon from './Icon';
import { themeSettings, styled, ThemeContext } from '../theme';

const LinkStyles = styled.a<StyleProps>`
  :hover,
  :active,
  :focus {
    text-decoration: underline;
  }
  text-decoration: none;
  display: inline-block;
  padding: 0.4rem 0.8rem;
  border-radius: ${themeSettings.borderRadius[8]};
  &.simple {
    padding: 0;
  }
  background-color: ${({ bg, theme }) => bg || theme.ui.link.background};
  color: ${({ fg, theme }) => fg || theme.ui.link.color};
  font-weight: ${props => (props.bg ? '300' : '500')};
  & > svg {
    margin-left: 1.2rem;
  }
`;

/**
 * !TODO: When StyledComponent gets updated
 * chek this again to remove duplication since a={Link} currently throws error
 */
const LinkStyles2 = styled(Link)<StyleProps>`
  :hover,
  :active,
  :focus {
    text-decoration: underline;
  }
  text-decoration: none;
  display: inline-block;
  padding: 0.4rem 0.8rem;
  border-radius: ${themeSettings.borderRadius[8]};
  &.simple {
    padding: 0;
  }
  background-color: ${({ bg, theme }) => bg || theme.ui.link.background};
  color: ${({ fg, theme }) => fg || theme.ui.link.color};
  font-weight: ${props => (props.bg ? '300' : '500')};
  & > svg {
    margin-left: 1.2rem;
  }
`;

const HighlightExternalLinkStyles = styled.a<StyleProps>`
  &:hover,
  &:active,
  &:focus {
    text-decoration: underline;
  }
  text-decoration: none;
  color: ${({ fg, theme }) => fg || theme.ui.link.color};
  background-color: ${({ bg, theme }) => bg || theme.ui.link.background};
  font-weight: 600;
  & > svg {
    margin-left: 1.2rem;
  }
  font-size: ${themeSettings.fontSize[24]};
`;

type StyleProps = {
  fg?: string;
  bg?: string;
};

const ExternalLink = ({ children, ...props }) => {
  const themeContext = useContext(ThemeContext);

  return (
    <LinkStyles {...props} target="_blank">
      {children}
      <Icon icon={faLongArrowRight} color={props.fg ?? themeContext.ui.link.icon.external.color} />
    </LinkStyles>
  );
};

const HighlightExternalLink = ({ children, ...props }) => {
  const themeContext = useContext(ThemeContext);

  return (
    <HighlightExternalLinkStyles {...props} target="_blank">
      {children}
      <Icon icon={faExternalLink} color={props.fg ?? themeContext.ui.link.icon.external.color} />
    </HighlightExternalLinkStyles>
  );
};

const SimpleExternalLink = ({ children, ...props }) => (
  <LinkStyles {...props} target="_blank" className="simple">
    {children}
  </LinkStyles>
);

const InternalLink = ({ children, ...props }) => {
  const themeContext = useContext(ThemeContext);

  return (
    <LinkStyles2 to={props.to} {...props}>
      {children}
      <Icon icon={faLongArrowRight} color={props.fg ?? themeContext.ui.link.icon.internal.color} />
    </LinkStyles2>
  );
};

export { ExternalLink, InternalLink, SimpleExternalLink, HighlightExternalLink };
