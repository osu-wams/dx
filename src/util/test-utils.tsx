import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render as testingLibraryRender } from '@testing-library/react';
import { Context as ResponsiveContext } from 'react-responsive';
import { themesLookup, defaultTheme } from '../theme/themes';
import { User } from '@osu-wams/lib';
import { mobile, desktop } from 'src/util/useMediaQuery';
import { RecoilRoot } from 'recoil';
import { userState } from 'src/state/application';
import { rest } from 'msw';
import { server } from 'src/mocks/server';

// Helper method to change the mock responses by MSW
export const alterMock = (api: string, mock: any) => {
  server.use(
    rest.get(api, async (req, res, ctx) => {
      return res(ctx.json(mock));
    })
  );
};

const { mockUser } = User;
export const authUserAudienceOverride = mockUser.userAudienceOverride;

export const mockGradUser = {
  ...mockUser.userGraduate,
  refreshFavorites: jest.fn(),
};

export const authUserClassification = mockUser.userClassification;

export const mockEmployeeUser = {
  ...mockUser.userEmployee,
  refreshFavorites: jest.fn(),
};

export const mockStudentEmployeeUser = {
  ...mockUser.userStudentEmployee,
  refreshFavorites: jest.fn(),
};

// The default undegraduate user with canvas opted in
export const authUser = {
  ...mockUser.user,
  refreshFavorites: jest.fn(),
};

// The default undegraduate user with canvas opted in
export const mockAdminUser = {
  ...mockUser.userAdmin,
  refreshFavorites: jest.fn(),
};

const renderWithUserContext = (
  ui,
  { user = authUser, initialStates = new Array(), ...options } = {}
) => {
  const Wrapper = (props) => {
    return (
      <RecoilRoot
        initializeState={(snap) => {
          snap.set(userState, user);
          initialStates.forEach((s: { state: any; value: any }) => snap.set(s.state, s.value));
        }}
      >
        <ThemeProvider theme={themesLookup[defaultTheme]} {...props} />
      </RecoilRoot>
    );
  };
  return testingLibraryRender(ui, { wrapper: Wrapper, ...options });
};

const renderWithAppContext = (ui, { initialStates = new Array(), ...options } = {}) => {
  const Wrapper = (props) => {
    return (
      <RecoilRoot
        initializeState={(snap) => {
          snap.set(userState, authUser);
          initialStates.forEach((s: { state: any; value: any }) => snap.set(s.state, s.value));
        }}
      >
        <ThemeProvider theme={themesLookup[defaultTheme]} {...props} />
      </RecoilRoot>
    );
  };
  return testingLibraryRender(ui, { wrapper: Wrapper, ...options });
};

const renderWithAllContexts = (
  ui,
  { user = authUser, isDesktop = false, initialStates = new Array(), ...options } = {}
) => {
  const Wrapper = (props) => {
    return (
      <RecoilRoot
        initializeState={(snap) => {
          snap.set(userState, user);
          initialStates.forEach((s: { state: any; value: any }) => snap.set(s.state, s.value));
        }}
      >
        <ThemeProvider theme={themesLookup[defaultTheme]}>
          <ResponsiveContext.Provider value={{ width: isDesktop ? desktop : mobile }} {...props} />
        </ThemeProvider>
      </RecoilRoot>
    );
  };
  return testingLibraryRender(ui, { wrapper: Wrapper, ...options });
};

const render = renderWithAllContexts;
// Pass a different user
// const { getByTestId } = renderWithUserContext(<Dashboard />, { user: authUser });

export { renderWithUserContext, renderWithAppContext, renderWithAllContexts, render };
