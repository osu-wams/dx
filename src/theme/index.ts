import { shadows, gradients, breakpoints, themeSettings } from './theme-settings';
import { Color } from './theme-colors';
import { ThemeContext, ThemeConfiguration as TC, styled } from './theme-interface';
import GlobalStyles from './GlobalStyles';
import { MainGridWrapper, MainGrid, SecondGridWrapper } from './grid/PageGrid';
import { Masonry } from './grid/Masonry';

export {
  shadows,
  gradients,
  breakpoints,
  GlobalStyles,
  themeSettings,
  Color,
  ThemeContext,
  styled,
  MainGridWrapper,
  MainGrid,
  SecondGridWrapper,
  Masonry
};

export type ThemeConfiguration = TC;
