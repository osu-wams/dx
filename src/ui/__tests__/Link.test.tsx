/* eslint-disable testing-library/no-node-access, testing-library/no-container */
import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithAllContexts as render } from 'src/util/test-utils';
import { Color } from '@osu-wams/theme';
import { ExternalLink, HighlightExternalLink, InternalLink, SimpleExternalLink, SimpleInternalLink, SimpleModalLink } from '../Link';

const Default = () => <ExternalLink href="https://oregonstate.edu">Default link</ExternalLink>;

const BackgroundColor = () => (
  <ExternalLink bg={Color['orange-400']} href="https://oregonstate.edu">
    Orange Background link
  </ExternalLink>
);

const TextColor = () => (
  <ExternalLink bg={Color['pine-400']} href="https://oregonstate.edu">
    Green Text link
  </ExternalLink>
);

const BothColors = () => (
  <ExternalLink bg={Color['orange-400']} fg={Color.white} href="https://oregonstate.edu">
    Orange Background with White text
  </ExternalLink>
);

const SimpleExtLink = () => (
  <SimpleExternalLink href="https://oregonstate.edu">
    Simple External Link Sample
  </SimpleExternalLink>
);

const InternalLinkBothColors = () => (
  <InternalLink bg={Color['orange-400']} fg={Color.white} to="https://oregonstate.edu">
    Orange Background with White text
  </InternalLink>
);

const SimpleIntLink = () => (
  <SimpleInternalLink to="https://oregonstate.edu">
    Simple Internal Link Sample
  </SimpleInternalLink>
);

const HighlightExtLink = () => (
  <HighlightExternalLink fg={Color['orange-400']} href="https://oregonstate.edu">
    Highlighted external link
  </HighlightExternalLink>
);

const SimpleModLink = () => (
  <SimpleModalLink>
    Simple Modal Link
  </SimpleModalLink>
);

// Snapshot tests to make sure our props are carried through
test('Default options', () => {
  const { container } = render(<Default />);
  expect(container).toMatchSnapshot();
});

test('Background color is orange-400', () => {
  const { container } = render(<BackgroundColor />);
  expect(container).toMatchSnapshot();
});

test('Text color is pine-400', () => {
  const { container } = render(<TextColor />);
  expect(container).toMatchSnapshot();
});

test('Background is orange-400, and text is white', () => {
  const { container } = render(<BothColors />);
  expect(container).toMatchSnapshot();
});

test('Make sure ExternalLink target="_blank" attribute is passed down', () => {
  render(<Default />);
  expect(screen.getByRole('link', { name: 'Default link' })).toHaveAttribute('target', '_blank');
});

test('Make sure SimpleExternalLink target="_blank" attribute is passed down', () => {
  render(<SimpleExtLink />);
  expect(screen.getByRole('link', { name: 'Simple External Link Sample' })).toHaveAttribute(
    'target',
    '_blank'
  );
});

test('Make sure SVG image is present', () => {
  const { container } = render(<TextColor />);
  const svg = container.querySelector('svg');

  expect(container).toContainElement(svg);
});

test('InternalLink background is orange-400 and text is white', () => {
  const { container } = render(<InternalLinkBothColors />);
  expect(container).toMatchSnapshot();
});

test('Test SimpleInternalLink props are passed down', () => {
  const { container } = render(<SimpleIntLink />);
  expect(container).toMatchSnapshot();
});

test('Highlighted text is orange', () => {
  const { container } = render(<HighlightExtLink />);
  expect(container).toMatchSnapshot();
});

test('Basic SimpleModalLink test', () => {
  const { container } = render(<SimpleModLink />);
  expect(container).toMatchSnapshot();
});
