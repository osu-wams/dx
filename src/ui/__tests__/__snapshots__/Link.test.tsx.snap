// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background color is orange-400 1`] = `
.c1 {
  color: #D73F09;
  background-color: transparent;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #D73F09;
  background-color: transparent;
  display: inline-block;
  padding: 4px 8px;
  background-color: #D73F09;
  font-weight: 500;
  border-radius: 8px;
}

.c0:hover,
.c0:active,
.c0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 > svg {
  margin-left: 1.2rem;
}

<div>
  <a
    class="c0"
    href="https://oregonstate.edu"
    target="_blank"
  >
    Orange Background link
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-long-arrow-right fa-w-14 c1"
      color="#D73F09"
      data-icon="long-arrow-right"
      data-prefix="fal"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M311.03 131.515l-7.071 7.07c-4.686 4.686-4.686 12.284 0 16.971L387.887 239H12c-6.627 0-12 5.373-12 12v10c0 6.627 5.373 12 12 12h375.887l-83.928 83.444c-4.686 4.686-4.686 12.284 0 16.971l7.071 7.07c4.686 4.686 12.284 4.686 16.97 0l116.485-116c4.686-4.686 4.686-12.284 0-16.971L328 131.515c-4.686-4.687-12.284-4.687-16.97 0z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;

exports[`Background is orange-400, and text is white 1`] = `
.c1 {
  color: #ffffff;
  background-color: transparent;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ffffff;
  background-color: transparent;
  display: inline-block;
  padding: 4px 8px;
  background-color: #D73F09;
  font-weight: 500;
  border-radius: 8px;
}

.c0:hover,
.c0:active,
.c0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 > svg {
  margin-left: 1.2rem;
}

<div>
  <a
    class="c0"
    href="https://oregonstate.edu"
    target="_blank"
  >
    Orange Background with White text
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-long-arrow-right fa-w-14 c1"
      color="#ffffff"
      data-icon="long-arrow-right"
      data-prefix="fal"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M311.03 131.515l-7.071 7.07c-4.686 4.686-4.686 12.284 0 16.971L387.887 239H12c-6.627 0-12 5.373-12 12v10c0 6.627 5.373 12 12 12h375.887l-83.928 83.444c-4.686 4.686-4.686 12.284 0 16.971l7.071 7.07c4.686 4.686 12.284 4.686 16.97 0l116.485-116c4.686-4.686 4.686-12.284 0-16.971L328 131.515c-4.686-4.687-12.284-4.687-16.97 0z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;

exports[`Default options 1`] = `
.c1 {
  color: #D73F09;
  background-color: transparent;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #D73F09;
  background-color: transparent;
  display: inline-block;
  padding: 4px 8px;
}

.c0:hover,
.c0:active,
.c0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 > svg {
  margin-left: 1.2rem;
}

<div>
  <a
    class="c0"
    href="https://oregonstate.edu"
    target="_blank"
  >
    Default link
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-long-arrow-right fa-w-14 c1"
      color="#D73F09"
      data-icon="long-arrow-right"
      data-prefix="fal"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M311.03 131.515l-7.071 7.07c-4.686 4.686-4.686 12.284 0 16.971L387.887 239H12c-6.627 0-12 5.373-12 12v10c0 6.627 5.373 12 12 12h375.887l-83.928 83.444c-4.686 4.686-4.686 12.284 0 16.971l7.071 7.07c4.686 4.686 12.284 4.686 16.97 0l116.485-116c4.686-4.686 4.686-12.284 0-16.971L328 131.515c-4.686-4.687-12.284-4.687-16.97 0z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;

exports[`Highlighted text is orange 1`] = `
.c2 {
  color: #D73F09;
  background-color: transparent;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #D73F09;
  background-color: transparent;
  display: inline-block;
  padding: 4px 8px;
}

.c0:hover,
.c0:active,
.c0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 > svg {
  margin-left: 1.2rem;
}

.c1 {
  padding: 0;
  font-weight: 600;
  font-size: 24px;
}

<div>
  <a
    class="c0 c1"
    href="https://oregonstate.edu"
    target="_blank"
  >
    Highlighted external link
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-external-link fa-w-16 c2"
      color="#D73F09"
      data-icon="external-link"
      data-prefix="fal"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M440,256H424a8,8,0,0,0-8,8V464a16,16,0,0,1-16,16H48a16,16,0,0,1-16-16V112A16,16,0,0,1,48,96H248a8,8,0,0,0,8-8V72a8,8,0,0,0-8-8H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V264A8,8,0,0,0,440,256ZM500,0,364,.34a12,12,0,0,0-12,12v10a12,12,0,0,0,12,12L454,34l.7.71L131.51,357.86a12,12,0,0,0,0,17l5.66,5.66a12,12,0,0,0,17,0L477.29,57.34l.71.7-.34,90a12,12,0,0,0,12,12h10a12,12,0,0,0,12-12L512,12A12,12,0,0,0,500,0Z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;

exports[`InternalLink background is orange-400 and text is white 1`] = `
.c1 {
  color: #ffffff;
  background-color: transparent;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ffffff;
  background-color: transparent;
  display: inline-block;
  padding: 4px 8px;
  background-color: #D73F09;
  font-weight: 500;
  border-radius: 8px;
}

.c0:hover,
.c0:active,
.c0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 > svg {
  margin-left: 1.2rem;
}

<div>
  <a
    bg="#D73F09"
    class="c0"
    fg="#ffffff"
    href="/https://oregonstate.edu"
  >
    Orange Background with White text
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-long-arrow-right fa-w-14 c1"
      color="#ffffff"
      data-icon="long-arrow-right"
      data-prefix="fal"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M311.03 131.515l-7.071 7.07c-4.686 4.686-4.686 12.284 0 16.971L387.887 239H12c-6.627 0-12 5.373-12 12v10c0 6.627 5.373 12 12 12h375.887l-83.928 83.444c-4.686 4.686-4.686 12.284 0 16.971l7.071 7.07c4.686 4.686 12.284 4.686 16.97 0l116.485-116c4.686-4.686 4.686-12.284 0-16.971L328 131.515c-4.686-4.687-12.284-4.687-16.97 0z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;

exports[`Test SimpleInternalLink props are passed down 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #D73F09;
  background-color: transparent;
  display: inline-block;
  padding: 0;
}

.c0:hover,
.c0:active,
.c0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 > svg {
  margin-left: 1.2rem;
}

<div>
  <a
    class="c0"
    href="/https://oregonstate.edu"
    padding="0"
  >
    Simple Internal Link Sample
  </a>
</div>
`;

exports[`Text color is pine-400 1`] = `
.c1 {
  color: #D73F09;
  background-color: transparent;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #D73F09;
  background-color: transparent;
  display: inline-block;
  padding: 4px 8px;
  background-color: #4A773C;
  font-weight: 500;
  border-radius: 8px;
}

.c0:hover,
.c0:active,
.c0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 > svg {
  margin-left: 1.2rem;
}

<div>
  <a
    class="c0"
    href="https://oregonstate.edu"
    target="_blank"
  >
    Green Text link
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-long-arrow-right fa-w-14 c1"
      color="#D73F09"
      data-icon="long-arrow-right"
      data-prefix="fal"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M311.03 131.515l-7.071 7.07c-4.686 4.686-4.686 12.284 0 16.971L387.887 239H12c-6.627 0-12 5.373-12 12v10c0 6.627 5.373 12 12 12h375.887l-83.928 83.444c-4.686 4.686-4.686 12.284 0 16.971l7.071 7.07c4.686 4.686 12.284 4.686 16.97 0l116.485-116c4.686-4.686 4.686-12.284 0-16.971L328 131.515c-4.686-4.687-12.284-4.687-16.97 0z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;
